src/tests/testCbcPointer.cc

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+/* DEBUG: section --    CbcPointer */
+
+#include "squid.h"
+#include "base/CbcPointer.h"
+#include "compat/cppunit.h"
+#include "unitTestMain.h"
+
+class TestCbcPointer : public CPPUNIT_NS::TestFixture
+{
+    CPPUNIT_TEST_SUITE(TestCbcPointer);
+    CPPUNIT_TEST(testConstruction);
+    CPPUNIT_TEST(testLogicOperators);
+    CPPUNIT_TEST(testCounting);
+    CPPUNIT_TEST_SUITE_END();
+
+protected:
+    void testConstruction();
+    void testLogicOperators();
+    void testCounting();
+};
+CPPUNIT_TEST_SUITE_REGISTRATION( TestCbcPointer );
+
+class Blob
+{
+    CBDATA_CLASS(Blob);
+public:
+    Blob() { ++Instances; }
+    ~Blob() { --Instances; }
+
+    static int Instances;
+};
+CBDATA_CLASS_INIT(Blob);
+
+int Blob::Instances = 0;
+
+void
+TestCbcPointer::testConstruction()
+{
+    Blob *nil = nullptr;
+    {
+        // default nil
+        CbcPointer<Blob> empty;
+        CPPUNIT_ASSERT_EQUAL(nil, empty.raw());
+        CPPUNIT_ASSERT_EQUAL(false, empty.set());
+        CPPUNIT_ASSERT_EQUAL(nil, empty.get());
+        // destruct of default-nil
+    }
+
+    {
+        // explicit nil
+        CbcPointer<Blob> empty(nullptr);
+        CPPUNIT_ASSERT_EQUAL(nil, empty.raw());
+        CPPUNIT_ASSERT_EQUAL(false, empty.set());
+        CPPUNIT_ASSERT_EQUAL(nil, empty.get());
+        // destruct of explicit-nil
+    }
+
+    {
+        auto expect = new Blob;
```

Comment:
There are a lot of _different_ "expectations" in this code. I recommend calling this pointer `raw`, `rawBlob`, or `blob` instead.

Reply:
This ("expect") is how unit tests name the variable holding the authoritative value(s) to test state against.